cli/azd/pkg/apphost/generate.go

```diff
@@ -274,14 +281,22 @@ func BicepTemplate(name string, manifest *Manifest, options AppHostOptions) (*me
 	for _, key := range genParametersKeys {
 		parameter := generator.bicepContext.InputParameters[key]
 		parameterMetadata := ""
-		if parameter.Default != nil && parameter.Default.Generate != nil {
-			pMetadata, err := inputMetadata(*parameter.Default.Generate)
-			if err != nil {
-				return nil, fmt.Errorf("generating input metadata for %s: %w", key, err)
+		var parameterDefaultValue *string
```

Comment:
I wonder if it's more readable if we can build the structs incrementally, which avoids local variables.

Here's what I mean:
```golang
input := genInput{Name: key, Secret: parameter.Secret, Type: parameter.Type}
if parameter.Default != nil {
	input.Value = parameter.Default.Value
}

autoGen := autoGenInput{genInput: input, MetadataType: azure.AzdMetadataTypeGenerate}
if parameter.Default != nil && parameter.Default.Generate != nil {
	pMetadata, err := inputMetadata(*parameter.Default.Generate)
	if err != nil {
		return nil, fmt.Errorf("generating input metadata for %s: %w", key, err)
	}
	autoGen.MetadataConfig = pMetadata
	// Note: azd is not checking or validating that Default.Generate and Default.Value are not both set.
	// The AppHost prevents this from happening by not allowing both to be set at the same time.
}

parameters = append(parameters, autoGen)
```

I think there's less indentations here. It also *subjectively* slightly feels more idiomatic go for me.

Reply:
BTW, one thing that kinda falls out by doing this -- it feels weird that `autoGenInput` is a struct that wraps `genInput`. This may be a result of the text/template context usage, but in my mind, an input has optional auto-gen properties.